src/modules/Product/Service.php

```diff
@@ -625,19 +633,50 @@ public function updatePromo(\Model_Promo $model, array $data = [])
         if (empty($products) && !is_null($products)) {
             $model->products = null;
         } else {
-            $model->products = json_encode($products);
+            $array = [];
```

Comment:
Can we name this var more descriptive than "array?"

Reply:
It is only to remove the "empty" value to make sure the the form get submitted when removing no values are clicked probally we should make it a function out of it as it will get used few times more ...

Reply:
I have created a function:
```
    public function removeEmptyValues($tobecleaned){
        if(is_null($tobecleaned)){
            return null;
        }
        $cleaned = [];
        foreach($tobecleaned as $value){
            if($value != ''){
                $cleaned[] = $value;
            }
        }
        if(empty($cleaned)){
            return null;
        }else{
            return $cleaned;
        }
    }
    ```
That does the task for us will also update the other locations